// WxWindows Patch
// Generated from TIP update on 6/5/2002
// Submitted 6/5/2002
// Robert Cavanaugh
// Description:
// the wxStrstr() method of WxChar does not work properly because the compare
// used is not checking the length. This resulted in erroneous matches for 
// strings in a text file. The corrected version uses Strncmp().
// I also added the capability to do searches backwards in a string and
// using case insensitivity.
diff -rbu2 Patched/wxWindows/include/wx/wxchar.h Tip/wxWindows/include/wx/wxchar.h
--- Patched/wxWindows/include/wx/wxchar.h	2002-06-05 13:38:48.000000000 -0600
+++ Tip/wxWindows/include/wx/wxchar.h	2002-04-05 03:43:34.000000000 -0700
@@ -4,7 +4,4 @@
 // Author:      Joel Farley
 // Modified by:
-//              Robert Cavanaugh:
-//              Added new functions for case sensitivity and direction
-//              of substring compares
 // Created:     1998/06/12
 // RCS-ID:      $Id: wxchar.h,v 1.77 2002/04/05 10:43:33 VZ Exp $
@@ -661,10 +658,4 @@
 #endif
 
-// Added 12/12/2001 Robert Cavanaugh
-// New string find definitions (not provided by any compiler)
-WXDLLEXPORT wxChar * wxStristr(const wxChar *haystack, const wxChar *needle);
-WXDLLEXPORT wxChar * wxStrrstr(const wxChar *haystack, const wxChar *needle);
-WXDLLEXPORT wxChar * wxStrirstr(const wxChar *haystack, const wxChar *needle);
-
 // under VC++ 6.0 isspace() returns 1 for 8 bit chars which completely breaks
 // the file parsing - this may be true for 5.0 as well, update #ifdef then
diff -rbu2 Patched/wxWindows/src/common/wxchar.cpp Tip/wxWindows/src/common/wxchar.cpp
--- Patched/wxWindows/src/common/wxchar.cpp	2002-06-05 13:48:02.000000000 -0600
+++ Tip/wxWindows/src/common/wxchar.cpp	2001-09-28 07:34:12.000000000 -0600
@@ -3,7 +3,5 @@
 // Purpose:     wxChar implementation
 // Author:      Ove Kåven
-// Modified by: Robert Cavanaugh
-//              Fixed bug in substring compare function
-//              added new substring compare functions
+// Modified by:
 // Created:     09/04/99
 // RCS-ID:      $Id: wxchar.cpp,v 1.29 2001/09/28 13:34:12 VZ Exp $
@@ -286,15 +284,9 @@
 }
 
-// Modified 12/12/2001 Robert Cavanaugh
-
 WXDLLEXPORT wxChar * wxStrstr(const wxChar *haystack, const wxChar *needle)
 {
-
-  // Need to compare with length, otherwise it can find the wrong string
   wxChar *fnd;
-  size_t len;
-  len = wxStrlen(needle)
   while ((fnd = wxStrchr(haystack, *needle))) {
-    if (!wxStrncmp(fnd, needle, len)) return fnd;
+    if (!wxStrcmp(fnd, needle)) return fnd;
     haystack = fnd + 1;
   }
@@ -503,54 +495,4 @@
 #endif
 
-// Added 21/12/2001 Robert Cavanaugh
-// New functions not provided by any default library
-
-// Find the first occurrence of the substring case insensitive
-WXDLLEXPORT wxChar * wxStristr(const wxChar *haystack, const wxChar *needle)
-{
-  wxChar *fnd;
-  size_t len;
-  len = wxStrlen(needle);
-  while ((fnd = wxStrchr(haystack, *needle)) != NULL)
-  {
-    if (!wxStrnicmp(fnd, needle, len)) return fnd;
-    haystack = fnd + 1;
-  }
-  return (wxChar *)NULL;
-}
-
-// Find the last occurrence of the substring case sensitive
-WXDLLEXPORT wxChar * wxStrrstr(const wxChar *haystack, const wxChar *needle)
-{
-  wxChar *ret = (wxChar *)NULL;
-  wxChar *fnd;
-  size_t len;
-  len = wxStrlen(needle);
-
-  while ((fnd = wxStrchr(haystack, *needle)) != NULL)
-  {
-    if (!wxStrncmp(fnd, needle, len)) ret = fnd;
-    haystack = fnd + 1;
-  }
-  return ret;
-}
-
-// Find the last occurrence of the substring case insensitive
-WXDLLEXPORT wxChar * wxStrirstr(const wxChar *haystack, const wxChar *needle)
-{
-  wxChar *ret = (wxChar *)NULL;
-  wxChar *fnd;
-  size_t len;
-  len = wxStrlen(needle);
-
-  while ((fnd = wxStrchr(haystack, *needle)) != NULL)
-  {
-    if (!wxStrnicmp(fnd, needle, len)) ret = fnd;
-    haystack = fnd + 1;
-  }
-  return ret;
-}
-
-
 #ifdef wxNEED_WX_TIME_H
 WXDLLEXPORT size_t   wxStrftime(wxChar *s, size_t max, const wxChar *fmt, const struct tm *tm)

 	  	 
